File: Code/petiapp/lib/models/request.dart
import 'package:petiapp/models/pet.dart';

class Request {
  Request({
    required this.sender,
    required this.receiver,
    required this.isAccepted,
    required this.isCompleted,
    required this.createdAt,
    required this.id,
  });

  Pet sender;
  Pet receiver;
  bool? isAccepted;
  bool isCompleted;
  String createdAt;
  String id;

  factory Request.fromMap(Map<String, dynamic> json) => Request(
        sender: Pet.fromMap(json["sender"]),
        receiver: Pet.fromMap(json["receiver"]),
        isAccepted: json["is_accepted"],
        isCompleted: json["is_completed"],
        createdAt: json["created_at"],
        id: json["id"],
      );
}

File: Code/server/app/models/Request.py
from datetime import datetime
from typing import Optional
from . import Pet
from odmantic import Model, Reference, Field

class Request(Model):
    sender: Pet = Reference()
    receiver: Pet = Reference()
    is_accepted: Optional[bool] = None
    is_completed: bool = False
    created_at: datetime = Field(default_factory=datetime.utcnow)

    class Config:
        collection = 'requests'
        parse_doc_with_default_factories = True